src/array/binary/mod.rs

```diff
@@ -94,6 +94,18 @@ impl<O: Offset> BinaryArray<O> {
             offset: self.offset + offset,
         }
     }
+
+    /// Sets the validity bitmap on this [`BinaryArray`].
+    /// # Panic
+    /// This function panics iff `validity.len() < self.len()`.
+    pub fn with_validity(&self, validity: Option<Bitmap>) -> Self {
+        if matches!(&validity, Some(bitmap) if bitmap.len() < self.len()) {
```

Comment:
```suggestion
        if matches!(&validity, Some(bitmap) if bitmap.len() != self.len()) {
```

it must be equal to keep the invariant stronger so that people can use `validity.len()` or `values.len()` when iterating item by item.

Reply:
> Very close; imo we should keep them equal to keep the invariant useful for people that iterate over one or the other without fear of going out of bounds.

I was just thinking if that could lead to problems. Agreed.